packages/foam-cli/src/commands/janitor.ts

```diff
@@ -0,0 +1,73 @@
+import { Command, flags } from '@oclif/command'
+import { initializeNoteGraph, generateLinkReferences, generateHeading, getKebabCaseFileName } from 'foam-core';
+import * as path from 'path';
+import { applyTextEdit } from '../utils/apply-text-edit';
+import { writeFileToDisk } from '../utils/write-file-to-disk';
+import { renameFile } from '../utils/rename-file';
+
+
+export default class Janitor extends Command {
+  static description = 'Updates link references and heading across all the markdown files in the given workspaces';
+
+  static examples = [
+    `$ foam-cli janitor path-to-foam-workspace
+Sucessfully generated link references and heading!
+`,
+  ]
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+  }
+
+  static args = [{ name: 'workspacePath' }]
+
+  async run() {
+    const { args, flags } = this.parse(Janitor)
+
+    const { workspacePath = './' } = args;
+
+    const foamWorkspaceDir = path.join(__dirname, workspacePath);
+
+    const graph = await initializeNoteGraph(foamWorkspaceDir);
+
+    const notes = graph.getNotes();
+
+    const fileWritePromises = notes.map(note => {
+      // Generate Heading 
+      const heading = generateHeading(note);
+
+      const fileWithHeading = heading ? applyTextEdit(note.source, heading) : note.source;
+
+      // Generate Link References
+      const linkRefTextEdits = generateLinkReferences(note, graph);
```

Comment:
For readability, I'd like to see a bit code grouping here. Since these operations are not dependent on each other, you could group them like:
```
  // get edits
  const heading = ...
  const definitions = ...

  //apply edits
  let file = note.source;
  file = heading ? applyTextEdit(file, heading) : file;
  file = definitions ? applyTextEdit(file, definitions) : file;
```

Reply:
Awesome! That does make the code more readable. I have pushed the change.